modules/fleet-provisioning/src/entry.c

```
@@ -285,18 +355,39 @@ static GglError update_iot_endpoints(void) {
 }
 
 GglError run_fleet_prov(FleetProvArgs *args, pid_t *pid) {
-    {
-        int config_dir;
-        GglError ret
-            = ggl_dir_open(ggcredentials_path, O_RDONLY, false, &config_dir);
-        if (ret != GGL_ERR_OK) {
-            GGL_LOGI("Could not open ggcredentials directory.");
-            return GGL_ERR_FAILURE;
-        }
-        (void) ggl_close(config_dir);
+    // Check the config to see if we need to provision
+    GglError ret = fetch_prov_status();
+    if (ret != GGL_ERR_OK) {
+        GGL_LOGI("Device already provisioned skipping.");
+        return ret;
+    }
+    ret = check_if_claim_cert(*args);
+    if (ret != GGL_ERR_OK) {
+        GGL_LOGE(
+            "Claim certs aren't provided as required for fleet-provisioning"
+        );
+        return ret;
+    }
+
+    if (args->out_cert_path != NULL) {
+        ggcredentials_path
+            = (GglBuffer) { .data = (uint8_t *) args->out_cert_path,
+                            .len = strlen(args->out_cert_path) };
     }
 
-    GglError ret = fetch_from_db(args);
+    int config_dir;
+    ret = ggl_dir_open(ggcredentials_path, O_RDONLY, false, &config_dir);
+    if (ret != GGL_ERR_OK) {
+        GGL_LOGI(
+            "Could not open %.*s directory.",
+            (int) ggcredentials_path.len,
+            ggcredentials_path.data
+        );
+        return GGL_ERR_FAILURE;
```

Should return the same error code from ret, instead of changing it to GGL_ERR_FAILURE. Also, should the above log line be error level?